pkg/chunk/storage/factory.go

```diff
@@ -25,6 +25,8 @@ const (
 	StorageEngineTSDB   = "tsdb"
 )
 
+type IndexClientFactoryFunc func() (chunk.IndexClient, error)
```

Comment:
Don't we need to pass the config here?

Reply:
no, the config and creation of index client(which means passing the config) would both be managed by the caller. They would just pass a closure basically.